File: frontend/src/api/security.ts
import { api } from "./client";

export type FindingSeverity = "critical" | "high" | "medium" | "low" | "info";
export type FindingType = "vulnerability" | "secret" | "static_analysis";

export interface SecurityFinding {
  id: string;
  project_id: string;
  type: FindingType;
  severity: FindingSeverity;
  title: string;
  description: string | null;
  ai_explanation: string | null;
  file_path: string | null;
  line_number: number | null;
  cve_id: string | null;
  remediation: string | null;
  resolved: boolean;
  detected_at: string;
}

export interface ScanJob {
  job_id: string;
  status: "queued" | "running" | "succeeded" | "failed";
}

export async function triggerScan(projectId: string): Promise<ScanJob> {
  const { data } = await api.post<ScanJob>("/v1/security/scan", null, {
    params: { project_id: projectId },
  });
  return data;
}

export async function triggerScanAll(): Promise<ScanJob> {
  const { data } = await api.post<ScanJob>("/v1/security/scan-all");
  return data;
}

export async function getFindings(projectId: string): Promise<SecurityFinding[]> {
  const { data } = await api.get<SecurityFinding[]>("/v1/security/findings", {
    params: { project_id: projectId },
  });
  return data;
}

File: backend/app/schemas/security.py
"""Security finding schemas.

`SecurityFindingCreate` mirrors docs/02_Implementation_Guide.md §9.2.
"""

import datetime

from pydantic import BaseModel, ConfigDict

from app.db.models import Severity
from app.schemas.job import JobEnvelope


class SecurityFindingCreate(BaseModel):
    type: str  # "vulnerability", "secret", "static_analysis"
    severity: Severity
    title: str
    description: str | None = None
    file_path: str | None = None
    line_number: int | None = None
    cve_id: str | None = None
    remediation: str | None = None


class SecurityFindingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    type: str
    severity: Severity
    title: str
    description: str | None = None
    ai_explanation: str | None = None
    file_path: str | None = None
    line_number: int | None = None
    cve_id: str | None = None
    remediation: str | None = None
    resolved: bool
    detected_at: datetime.datetime


class ScanResponse(JobEnvelope):
    pass


class SecurityClearRead(BaseModel):
    """Response for DELETE /security/findings (v1.17.18.5, audit2 C5)."""

    deleted: int
